### ISSUE TITLE
question about inconsistent results for F1-score and PR-AUC in "PAAD" and "ROAR" Papers

### ISSUE THREAD

**Issue by flyflyly** [andreschreiber/ROAR#2]
Hello,

I have noticed an inconsistency in the reported results between the "PAAD" and "ROAR" papers.
The results table in "PAAD" as following,
![image](https://github.com/user-attachments/assets/e57ac4a5-6b3b-4529-84b2-7fcff4e3b963)
The results table in "ROAR" as following,
![image](https://github.com/user-attachments/assets/50d97499-f096-4556-9a1f-a122ec8b2afa)

In the results table from the "PAAD" paper, the F1-score and PR-AUC for PAAD are 0.6453 and 0.8281, respectively. However, in the results table from the "ROAR" paper, the F1-score and PR-AUC values for PAAD are different. Additionally, it seems that the results reported in the "PAAD" paper are better than those reported in the "ROAR" paper.

Could you please clarify why there is a discrepancy between these reported results?
Thank you for your help!

**Comment by andreschreiber** [comment]
Absolutely!

The primary reason is that using a different computer to train the models leads to different results (even if the same seed is used); different computers were used to generate the results in the PAAD paper and ROAR paper, so that will lead to slightly different results. Likewise, using different versions of PyTorch can also potentially cause different results for training. All results in the ROAR paper (including baselines) were generated using the same computer and conda environment to ensure consistency in that regard.

When I tried to train PAAD using the PAAD repo, I did not get the same result as mentioned in the PAAD paper (it had a lower PR-AUC than in the paper). This is due to the different computer that was used, and I think using a newer PyTorch version can lead to differences in results as well.

In addition, to account for factors like the variation in metric values when training on different seeds/computers, in the ROAR paper we trained 5 models (the results in the PAAD paper only used 1 training). Each training had a different random seed, but the models all used the same seeds; i.e., we trained ROAR on 5 seeds, and we also trained PAAD (as well as all other models) on the same five seeds (those seeds were just randomly chosen before running the experiments).

If you'd like to use the models that have the metrics in the paper, feel free to use the saved model weights linked in the README file. Evaluation of that model should give the results of "Best Model" in the table above. Likewise, I think the PAAD model from the original PAAD paper has its weights stored on Box linked on the PAAD repo.

Please don't hesitate to ask if you have any additional questions!

**Comment by flyflyly** [comment]
Thank you so much for your detailed explanation! I completely understand now how the variations in hardware, PyTorch versions, and random seeds can lead to different results.
